tests/devices/unit_tests/test_xspress3mini_OA.py

```diff
@@ -0,0 +1,45 @@
+import asyncio
+
+import pytest
+from bluesky import plan_stubs as bps
+from bluesky.run_engine import RunEngine
+from ophyd.status import Status
+from ophyd_async.core import DeviceCollector, set_sim_value
+
+from dodal.devices.ophyd_async_xspress3_mini.xspress3_mini_oa import (
+    AcquireState,
+    DetectorState,
+    Xspress3Mini,
+)
+
+
+def get_bad_status() -> Status:
+    status = Status("get_bad_status")
+    status.set_exception(Exception)
+    return status
+
+
+@pytest.fixture
+async def fake_xspress3mini(prefix: str = "huh") -> Xspress3Mini:
+    async with DeviceCollector(sim=True):
+        fake_xspress3mini = Xspress3Mini(prefix, "Xspress3Mini")
+    return fake_xspress3mini
+
+
+async def test_arm_success_on_busy_state_OA(fake_xspress3mini: Xspress3Mini):
+    # set_sim_value(fake_xspress3mini.detector_state,DetectorState.IDLE)
+    set_sim_value(fake_xspress3mini.detector_state, DetectorState.ACQUIRE)
+    status = await fake_xspress3mini.arm()
+    assert status.done is False
+    set_sim_value(fake_xspress3mini.acquire, AcquireState.DONE)
+    await asyncio.sleep(0.1)
+    assert status.done is True
+
+    set_sim_value(fake_xspress3mini.detector_state, DetectorState.IDLE)
+    with pytest.raises(TimeoutError):
+        await fake_xspress3mini.stage()
```

Comment:
Should: It seems like the bits above are testing that the detector arms when all is good and this lines that the detector doesn't stage when all is not good, the stage timesout, could they be split into two different tests?